### DISCUSSION TITLE
Add support to hook into the lock screen next and previous buttons in Media Element

### DISCUSSION THREAD

**Discussion by JohnStrudwick** [CommunityToolkit/Maui#2147] [New Feature Discussions]
I notice that the next and previous buttons are greyed out when using Media Element and the metadata features. Most other programs seem to have access to these. I would like to be able to enable them and hook into the pressed event for these so that I could hook up playlists etc.
![image](https://github.com/user-attachments/assets/cb35e8b0-bc00-4e12-adec-11e8477de277)

**Comment by ne0rrmatrix** [comment]
Which API's are you on which platform? And which version of MediaElement are you using? Yes I agree they need to be added but knowing which version you are on will help me discover why the previous button is not working. ATM on some platforms it will just start current video/music from the beginning. It is not greyed out for all platforms and devices. 

I have been planning on adding previous and next but I have been working on adding other things first. I am pretty much done with other things now so I will start working on this. but I need ppl to upvote this to show support. 

To add this support I imagine changing the behavior of media element from replacing the media item when you change sources to having an add/remove and clear sources options? And/or add list of items. I don't think we should do anything but add the capabilities to do this and allow developers to do all the management of playlists. But simply adding support to add/remove/update and delete items in the list would be best. It can be implemented on all platforms easily. They all have native support for this feature.

This would be a breaking change but to mitigate any issues we could support as a default option the current behaviour as is and have an add/remove/update/delete option with the default being replace if nothing else is selected? That would maintain current behaviour if developer makes no changes to their code but would allow us to update and add the new options.

**Reply by JohnStrudwick** [reply]
Hi. I am on MediaElement 4.0.1, toolkitMaui 9.0.2, toolkitMvvm 8.2.2, visual studio 17.8.5 and I am on iPhone (IOS 17.4.1)/iPad(IOS 16.6.1) and Windows (11) - I haven't looked at Android yet as iPhone, iPad & Windows are my primary targets. Windows and iPhone/iPad synchronise using Dropbox with some help from me as it's automatic on Windows but on iPhone/iPad I do the synchronising !

I agree to changing the behavior of media element from replacing the media item when you change sources to having an add/remove and clear sources options and/or add list of items. And yes let the developer handle the playlist.

Let me know if you need more information.

While on metadata, please note that I did also put a request in for the graphic to come from a stream or local file, not just a URL :)

**Reply by ne0rrmatrix** [reply]
I just put up a proposal for adding metadata support for artwork using local files and embedded resources. https://github.com/CommunityToolkit/Maui/issues/2151
